atc/api/accessor/accessor.go

```diff
@@ -135,18 +135,18 @@ func (a *access) rolesForTeam(auth atc.TeamAuth) []string {
 
 		// backwards compatibility for allow-all-users
 		if len(userAuth) == 0 && len(groupAuth) == 0 {
-			roles = append(roles, role)
+			roleSet[role] = true
 		}
 
 		for _, user := range userAuth {
 			if userID != "" {
 				if strings.EqualFold(user, fmt.Sprintf("%v:%v", connectorID, userID)) {
-					roles = append(roles, role)
+					roleSet[role] = true
 				}
 			}
 			if userName != "" {
 				if strings.EqualFold(user, fmt.Sprintf("%v:%v", connectorID, userName)) {
-					roles = append(roles, role)
+					roleSet[role] = true
```

Comment:
I think this one is not tested either

Reply:
yeah i think some of these tests got dropped when we moved this stuff around. re-added!